shared-bindings/pixelbuf/PixelBuf.c

```diff
@@ -0,0 +1,506 @@
+              (mp_obj_t)&pixelbuf_pixelbuf_set_brightness_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+void pixelbuf_recalculate_brightness(pixelbuf_pixelbuf_obj_t *self) {
+    for (uint i = 0; i < self->bytes; i++) {
+       self->buf[i] = self->rawbuf[i] * self->brightness;
+    }
+}
+
+//|   .. attribute:: buf
+//|
+//|     The bytearray of pixel data after brightness adjustment
+//|
+STATIC mp_obj_t pixelbuf_pixelbuf_obj_get_buf(mp_obj_t self_in) {
+    pixelbuf_pixelbuf_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_bytearray_by_ref(self->bytes, self->buf);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(pixelbuf_pixelbuf_get_buf_obj, pixelbuf_pixelbuf_obj_get_buf);
+
+const mp_obj_property_t pixelbuf_pixelbuf_buf_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&pixelbuf_pixelbuf_get_buf_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: byteorder
+//|
+//|     byteorder for the buffer (read-only)
+//|
+STATIC mp_obj_t pixelbuf_pixelbuf_obj_get_byteorder(mp_obj_t self_in) {
+    pixelbuf_pixelbuf_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int_from_uint(self->byteorder);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(pixelbuf_pixelbuf_get_byteorder_obj, pixelbuf_pixelbuf_obj_get_byteorder);
+
+const mp_obj_property_t pixelbuf_pixelbuf_byteorder_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&pixelbuf_pixelbuf_get_byteorder_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: len
+//|
+//|     The size of the pixel buffer in pixels (read-only)
+//|
+STATIC mp_obj_t pixelbuf_pixelbuf_obj_get_len(mp_obj_t self_in) {
+    pixelbuf_pixelbuf_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int_from_uint(self->pixels);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(pixelbuf_pixelbuf_get_len_obj, pixelbuf_pixelbuf_obj_get_len);
+
+const mp_obj_property_t pixelbuf_pixelbuf_len_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&pixelbuf_pixelbuf_get_len_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+
+//|   .. attribute:: rawbuf
+//|
+//|     The raw bytearray of pixel data if provided during construction.
+//|
+STATIC mp_obj_t pixelbuf_pixelbuf_obj_get_rawbuf(mp_obj_t self_in) {
+    pixelbuf_pixelbuf_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    if (self->rawbuf == NULL) {
+        return mp_const_none;
+    }
+    return mp_obj_new_bytearray_by_ref(self->bytes, self->rawbuf);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(pixelbuf_pixelbuf_get_rawbuf_obj, pixelbuf_pixelbuf_obj_get_rawbuf);
+
+const mp_obj_property_t pixelbuf_pixelbuf_rawbuf_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&pixelbuf_pixelbuf_get_rawbuf_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: size
```

Comment:
Why do you need this? Can't it be computed from the len and pixel order?

Reply:
I honestly have no good explanation for this one.  I was probably thinking along the lines of "if i'm working with bytes..." But I'm about to purge that code too.